.github/workflows/ci.yml

```
@@ -0,0 +1,352 @@
+name: CI 
+
+on:
+  push:
+    paths:
+      - '.github/workflows/dart_ci.yml'
+      - 'discovery_engine/**'
+      - 'discovery_engine_core/**'
+      - 'job_runner.sh'
+      - '.github/workflows/rust_ci.yml'
+      - 'discovery_engine_core/**'
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+env:
+  DART_VERSION: '2.14.4'
+  DART_WORKSPACE: ${{ github.workspace }}/discovery_engine
+  RUST_WORKSPACE: ${{ github.workspace }}/discovery_engine_core
+  RUST_NIGHTLY: nightly-2021-09-09
+  RUSTFLAGS: "-D warnings"
+  RUSTDOCFLAGS: "-D warnings"
+
+
+jobs:
+  selection:
+    runs-on: ubuntu-latest
+    # Set job outputs to values from filter step
+    outputs:
+      dart: ${{ steps.filter.outputs.dart }}
+      rust: ${{ steps.filter.outputs.rust }}
+    steps:
+      # For pull requests it's not necessary to checkout the code
+      - uses: dorny/paths-filter@v2.10.2
+        id: filter
+        with:
+          filters: |
+            dart:
+              - '.github/workflows/dart_ci.yml'
+              - 'discovery_engine/**'
+              - 'discovery_engine_core/**'
+              - 'job_runner.sh'
+            rust:
+              - '.github/workflows/rust_ci.yml'
+              - 'discovery_engine_core/**'
+
+  # DART CI 
+  dart-format:
+    name: dart-format
+    needs: selection
+    if: ${{ needs.changes.outputs.dart == 'true' }}
```

```suggestion
    if: ${{ needs.selection.outputs.dart == 'true' }}
```
or do I overlooked something?

Same for the other ifs.

---

Yes, that is correct. The `changes` and `test` are from my testing, but I updated them to `rust` and `dart` respectively.

---

How can it work if we are looking for a key in `needs` that we don't specify?